app/services/queue_listener.rb

```diff
@@ -101,4 +101,26 @@ def update_member(params)
       raise Error.new("Member update failed with #{res.parsed_response["errors"]}!")
     end
   end
+
+  def action_params(params)
+    country = params[:params][:country]
+    raise Error.new("Generic action processing attempted without a country in the request.") unless country.present?
+    if country == "United States" && ENV['BYPASS_ZIP_VALIDATION']
+      raise Error.new("Your default US zip code has not been set!") unless ENV['DEFAULT_US_ZIP'].present?
+      params[:params][:postal] = verify_zip(params[:params][:postal])
+    end
+    params
+  end
```

Comment:
I think this whole logic belongs in a different service class. I think the QueueListener ideally should only know "what" needs to be done, and ask for other classes to do it. We're already using this pattern with PageCreator and ActionCreator. 

I know this is not a simple change to do since it will require some refactoring. But eventually I strongly believe we should move in that direction.